source/PlayerInfo.cpp

```diff
@@ -708,27 +716,35 @@ int64_t PlayerInfo::Salaries() const
 
 
 
-// Calculate the daily maintenance cost for all ships and in cargo outfits.
-int64_t PlayerInfo::Maintenance() const
+// Calculate the daily maintenance cost and generated income for all ships and in cargo outfits.
+void PlayerInfo::MaintenanceAndReturns(int64_t &maintenance, int64_t &assetReturns) const
```

Comment:
Return a pair or struct rather than mutate input parameters.

(Not sure what would be a good name for the struct; the only benefit it would have over a std::pair would be named fields instead of `first` and `second`.)

A third option is to parameterize the function body by the strings and move it to a private / static helper function. Then you call it for either maintenance or generated income, and it just returns the `int64_t` of interest.

Reply:
I considered a pair, but the `first` and `second` fields are not really clear in the callers context. Using a struct is nice though, so I added a struct here.

Used the name `FleetBalance` for the struct (inspired by financial balance sheets). Not a great name, but the best I could come up with.

I prefer not to loop through the list of ships twice, so I didn't go for the parameterized function.